File: app/src/test/java/com/huynd/skyobserver/utils/ConstantsTest.java
package com.huynd.skyobserver.utils;

import org.junit.Test;

import java.lang.reflect.Constructor;

import static junit.framework.Assert.assertTrue;
import static junit.framework.Assert.fail;

/**
 * Created by HuyND on 8/13/2017.
 */

public class ConstantsTest {
    @Test
    public void testPrivateConstructor() {
        try {
            Constructor<Constants> constructor =
                    Constants.class.getDeclaredConstructor(new Class[0]);
            constructor.setAccessible(true);

            // should throws AssertionError here
            constructor.newInstance(new Object[0]);
            fail("This line is never reached!");
        } catch (Throwable e) {
            assertTrue("This line is reached!", true);
        }
    }
}

File: app/src/test/java/com/huynd/skyobserver/utils/ConstantsTest.kt
package com.huynd.skyobserver.utils

import org.junit.Assert.assertTrue
import org.junit.Assert.fail
import org.junit.Test

/**
 * Created by HuyND on 5/25/2019.
 */

class ConstantsTest {
    @Test
    fun testPrivateConstructor() {
        try {
            val constructor = Constants::class.java.getDeclaredConstructor()
            constructor.isAccessible = true

            // should throws AssertionError here
            constructor.newInstance()
            fail("This line is never reached!")
        } catch (e: Throwable) {
            assertTrue("This line is reached!", true)
        }
    }
}
